rotkehlchen/exchanges/htx.py

```diff
@@ -188,24 +221,235 @@ def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
+                coin = asset_from_htx(movement['currency'])
+            except UnknownAsset:
+                log.error(f'Could not read assets from trade pair {movement["currency"]}. Skipping...')  # noqa: E501
+                continue
+
+            try:
+                movements.append(AssetMovement(
+                    timestamp=timestamp,
+                    location=Location.HTX,
+                    category=AssetMovementCategory.DEPOSIT if movement['type'] == 'deposit' else AssetMovementCategory.WITHDRAWAL,  # noqa: E501
+                    address=movement['address'],
+                    transaction_id=movement['tx-hash'],
+                    asset=coin,
+                    amount=deserialize_asset_amount(movement['amount']),
+                    fee_asset=coin,
+                    fee=deserialize_fee(movement['fee']) if movement['fee'] != 0 else Fee(ZERO),
+                    link=str(movement['id']),
+                ))
+            except (DeserializationError, KeyError) as e:
+                msg = str(e)
+                if isinstance(e, KeyError):
+                    msg = f'Missing key {msg}'
+
+                log.error(f'{e} when reading movement for HTX deposit {movement}. Skipping...')
+                continue
+
+        return movements
+
     def query_online_deposits_withdrawals(
             self,
             start_ts: Timestamp,
             end_ts: Timestamp,
     ) -> list[AssetMovement]:
         """Query deposits and withdrawals sequentially"""
-        return []  # TODO: Implement this in another PR
+        new_movements = self._query_deposits_withdrawals(
+            start_ts=start_ts,
+            end_ts=end_ts,
+            query_for=AssetMovementCategory.DEPOSIT,
+        )
+        new_movements.extend(
+            self._query_deposits_withdrawals(
+                start_ts=start_ts,
+                end_ts=end_ts,
+                query_for=AssetMovementCategory.WITHDRAWAL,
+            ),
+        )
+        return new_movements
 
     def query_online_margin_history(
             self,
-            start_ts: Timestamp,  # pylint: disable=unused-argument
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list['MarginPosition']:
+        return []  # noop for htx
+
+    def query_online_income_loss_expense(
+            self,
+            start_ts: Timestamp,
             end_ts: Timestamp,
-    ) -> list[MarginPosition]:
-        return []  # TODO: Implement this in another PR
+    ) -> list['HistoryEvent']:
+        return []  # noop for htx
 
     def query_online_trade_history(
             self,
             start_ts: Timestamp,
             end_ts: Timestamp,
     ) -> tuple[list[Trade], tuple[Timestamp, Timestamp]]:
-        return [], (start_ts, end_ts)  # TODO: Implement this in another PR
+        """
+        Query trades from HTX in the spot category.
+        The API is limited to query trades up to 120 days into the past. We can query time ranges
```

Comment:
```suggestion
        The API is limited to query trades up to 120 days into the past. We can query time ranges
```

Are you 100% certain? This is extremely small. you would need to add a link to where in the docs this is determined so that we can also tell the users. Have you checked if ccxt does not use any other endpoints? Perhaps you are missing something

Reply:
Checked ccxt, it's the same endpoint, updated the link.